From 161eb6100d6f75f0a0df6da3d19da7fe842655c1 Mon Sep 17 00:00:00 2001
From: Soonil Nagarkar <sooniln@google.com>
Date: Fri, 25 Oct 2024 10:41:26 -0700
Subject: [PATCH] Add permissions check to isInSignificantPlace()

Require the ACCESS_FINE_LOCATION permission for this API.

Flag: EXEMPT bugfix
Bug: 374257207
Test: manual
(cherry picked from commit 389e4cf737571bf0c438239400295fd229acc3e8)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:f085fccf0bb7963962ae56618c5b3a1762d3aeab)
Merged-In: I1c5ef857c750dbbd0577f5aadc43e9b6122de790
Change-Id: I1c5ef857c750dbbd0577f5aadc43e9b6122de790
---
 core/java/android/app/trust/ITrustManager.aidl                | 1 +
 core/java/android/app/trust/TrustManager.java                 | 1 +
 .../java/com/android/server/trust/TrustManagerService.java    | 4 ++++
 3 files changed, 6 insertions(+)

diff --git a/core/java/android/app/trust/ITrustManager.aidl b/core/java/android/app/trust/ITrustManager.aidl
index 740f5932f902..730bb73da3bb 100644
--- a/core/java/android/app/trust/ITrustManager.aidl
+++ b/core/java/android/app/trust/ITrustManager.aidl
@@ -41,5 +41,6 @@ interface ITrustManager {
     void unlockedByBiometricForUser(int userId, in BiometricSourceType source);
     void clearAllBiometricRecognized(in BiometricSourceType target, int unlockedUser);
     boolean isActiveUnlockRunning(int userId);
+    @EnforcePermission("ACCESS_FINE_LOCATION")
     boolean isInSignificantPlace();
 }
diff --git a/core/java/android/app/trust/TrustManager.java b/core/java/android/app/trust/TrustManager.java
index 88d4d691cd97..1ef83cdf3f85 100644
--- a/core/java/android/app/trust/TrustManager.java
+++ b/core/java/android/app/trust/TrustManager.java
@@ -305,6 +305,7 @@ public class TrustManager {
      *
      * @hide
      */
+    @RequiresPermission(Manifest.permission.ACCESS_FINE_LOCATION)
     public boolean isInSignificantPlace() {
         try {
             return mService.isInSignificantPlace();
diff --git a/services/core/java/com/android/server/trust/TrustManagerService.java b/services/core/java/com/android/server/trust/TrustManagerService.java
index 3138a9ea9426..b59b002d720b 100644
--- a/services/core/java/com/android/server/trust/TrustManagerService.java
+++ b/services/core/java/com/android/server/trust/TrustManagerService.java
@@ -20,6 +20,7 @@ import static android.service.trust.GrantTrustResult.STATUS_UNLOCKED_BY_GRANT;
 import static android.service.trust.TrustAgentService.FLAG_GRANT_TRUST_TEMPORARY_AND_RENEWABLE;
 
 import android.Manifest;
+import android.annotation.EnforcePermission;
 import android.annotation.NonNull;
 import android.annotation.Nullable;
 import android.annotation.UserIdInt;
@@ -1868,8 +1869,11 @@ public class TrustManagerService extends SystemService {
             }
         }
 
+        @EnforcePermission(Manifest.permission.ACCESS_FINE_LOCATION)
         @Override
         public boolean isInSignificantPlace() {
+            super.isInSignificantPlace_enforcePermission();
+
             if (android.security.Flags.significantPlaces()) {
                 mSignificantPlaceServiceWatcher.runOnBinder(
                         binder -> ISignificantPlaceProvider.Stub.asInterface(binder)
-- 
2.48.1.262.g85cc9f2d1e-goog

